policy-engine/src/graph.rs

```diff
@@ -95,8 +95,11 @@ pub(crate) fn index(req: HttpRequest) -> Box<dyn Future<Item = HttpResponse, Err
 
             if let Err(e) = &r {
                 error!(
-                    "Error serving request with parameters '{:?}': {}",
-                    req.query_string(),
+                    "Error serving request '{}' from '{}': {:?}",
+                    format!("{:?}", &req).replace("\n", " ").replace("\t", " "),
+                    &req.peer_addr()
+                        .map(|addr| addr.to_string())
+                        .unwrap_or("<not available>".into()),
```

Comment:
this is not the only type of 400 error we generate, right?  I'd like to have User-Agents on all the errors we spit out when returning a 400 response.

Reply:
Following the chain of combinators like `and_then`, `map`, etc., you'll be able to notice that this  `then`, in which we log the error, is unconditionally attached at the end, and is thus applies to all responses which contain an error regardless of the specific type. This `index` function is used as the handler for the `v1/graph` endpoint, so it handles all errors going out of that. If you can generate a request which leads to an error and is not covered by this log output I'll consider it a bug.